src/parser/priest/shadow/modules/features/DarkThoughts.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import SPELLS from 'common/SPELLS/index';
+import SpellLink from 'common/SpellLink';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import { When, ThresholdStyle } from 'parser/core/ParseResults';
+import { t } from '@lingui/macro';
+import { formatPercentage } from 'common/format';
+import EventHistory from 'parser/shared/modules/EventHistory';
+
+// Example log: /report/JAPL1zpDfN7W8wck/33-Heroic+The+Council+of+Blood+-+Kill+(5:46)/Mayrim/standard/statistics
+class DarkThoughts extends Analyzer {
+  static dependencies = {
+    eventHistory: EventHistory,
+  };
+  protected eventHistory!: EventHistory;
+
+  procsGained: number = 0;
+  procsUsed: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.DARK_THOUGHT_BUFF), this.onBuffApplied);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.DARK_THOUGHT_BUFF), this.onBuffRemoved);
+  }
+
+  onBuffApplied() {
+    this.procsGained += 1; // Add a proc to the counter
+  }
+
+  onBuffRemoved() {
+    if (!this.eventHistory.last(1, 100, Events.cast.by(SELECTED_PLAYER).spell(SPELLS.MIND_BLAST))) { // If MB is not instant, it's not a proc
+      return;
+    }
+
+    this.procsUsed += 1;
+  }
+
+  getProcsWasted() {
+    return this.procsGained - this.procsUsed;
+  }
+
+  get suggestionThresholds() {
+    return {
+      actual: this.getProcsWasted() / this.procsGained,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 0.1,
+      },
+      style: ThresholdStyle.PERCENTAGE,
+    };
+  }
+
+  suggestions(when: When) {
+    when(this.suggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => suggest(<>You wasted {formatPercentage(actual)}% of <SpellLink id={SPELLS.DARK_THOUGHTS.id} /> procs. </>)
+          .icon(SPELLS.DARK_THOUGHTS.icon)
+          .actual(
+            t({
+              id:'priest.shadow.suggestions.darkThoughts.efficiency',
+              message: `Wasted ${formatPercentage(actual)}% of Dark Thought procs. Remember that the proc allows you to cast Mind Blast instantly during Mind Flay, Mind Sear, or while moving and should never be left unused.`
```

Comment:
Your translated suggestion has more info than the actual suggestion text. Might want to add that second sentence to the actual suggestion above there.

Keep in mind that each suggestion should aim to explain what you did, what you should have done, and why.

Reply:
Thanks, I hadn't really ever known what was required for a suggestion so good to know. 👍

Reply:
Rewrote the suggestion based on your recommendations. Also switched over to using actual values of procs instead of percentage because it's more relevant than a random %.

![image](https://user-images.githubusercontent.com/5396389/103047668-3a27eb80-455a-11eb-8dff-790d3a90906d.png)